### PR TITLE
created spiral matrix #11022

### PR DESCRIPTION
**Pull request by MD571**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: data_structures/arrays/sprialmatrix.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+def spiralorder(matrix: list[list[int]]) -> list[int]:
+    """
+    Get the elements of a 2D matrix in spiral order.
+
+    Args:
+    matrix (list[list[int]]): The input 2D matrix.
+
+    Returns:
+    List[int]: A list containing the elements of the matrix in spiral order.
+
```
Comment (imSanko) [inline data_structures/arrays/sprialmatrix.py:10]: ______________________ [doctest] sprialmatrix.spiralorder ______________________
004 
005     Args:
006     matrix (list[list[int]]): The input 2D matrix.
007 
008     Returns:
009     List[int]: A list containing the elements of the matrix in spiral order.
010 
011     Example:
012     >>> a = [[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12], [13, 14, 15, 16]]
013     >>> spiralOrder(a)

File: data_structures/arrays/sprialmatrix.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    Example:
+    >>> a = [[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12], [13, 14, 15, 16]]
+    >>> spiralOrder(a)
+    [1, 2, 3, 4, 8, 12, 16, 15, 14, 13, 9, 5, 6, 7, 11, 10]
```
Comment (imSanko) [inline data_structures/arrays/sprialmatrix.py:14]: Have you tried with python -m doctest -v data_structures/arrays/sprialmatrix.py

Comment (MD571) [reply data_structures/arrays/sprialmatrix.py:14]: yes if build is successful then remaining two are not successful

**Comment by cclauss** [comment]
Duplicate of matrix/spiral_print.py

Thank you for your contribution to TheAlgorithms/Python but this has been a __very busy HacktoberFest__ so we will be closing all new pull requests until the end of October so we can properly review approximately 200 open pull requests.
### We hope you will try to contribute again after October 31st.

**Pull request closed without merging**